patterns/ecs-v1/linux-syslog

```diff
@@ -1,16 +1,16 @@
 SYSLOG5424PRINTASCII [!-~]+
 
-SYSLOGBASE2 (?:%{SYSLOGTIMESTAMP:timestamp}|%{TIMESTAMP_ISO8601:timestamp8601}) (?:%{SYSLOGFACILITY} )?%{SYSLOGHOST:logsource}+(?: %{SYSLOGPROG}:|)
-SYSLOGPAMSESSION %{SYSLOGBASE} (?=%{GREEDYDATA:message})%{WORD:pam_module}\(%{DATA:pam_caller}\): session %{WORD:pam_session_state} for user %{USERNAME:username}(?: by %{GREEDYDATA:pam_by})?
+SYSLOGBASE2 (?:%{SYSLOGTIMESTAMP:timestamp}|%{TIMESTAMP_ISO8601:timestamp8601}) (?:%{SYSLOGFACILITY} )?%{SYSLOGHOST:event.provider}+(?: %{SYSLOGPROG}:|)
+SYSLOGPAMSESSION %{SYSLOGBASE} (?=%{GREEDYDATA:message})%{WORD:log.syslog.pam.module}\(%{DATA:log.syslog.pam.caller}\): session %{WORD:log.syslog.pam.session_state} for user %{USERNAME:log.syslog.pam.user}(?: by %{GREEDYDATA})?
```

Comment:
not sure how much `SYSLOGPAMSESSION` is used - we likely have to keep it around, 
do we use `log.syslog.pam.` namespace or move it out of the syslog-one `log.pam.` (PAM uses syslog)?

given a sample PAM message:
`Jul 14 13:36:03 precision pkexec: pam_unix(polkit-1:session): session opened for user root by (uid=1001)`
the `pam_by` capture seemed 'unfinished' (`"pam_by": "(uid=1001)"`) so I left it out, do we attempt to change parsing or is it okay to parse as before for ecs?
have intentionally restrained myself from changing any patterns but some

and if we parse it's a bit :weight_lifting: coming up with a good name: 
`parse_by`  -> `log.syslog.pam.authenticator` (and maybe `username`  -> `log.syslog.pam.authanticated_user`)

Reply:
I think this is probably a question for @elasitc/ecs

Reply:
hmm we can not ping a team cross repo even with the typo fixed // cc @webmat @ebeahan

Reply:
Thanks for the ping 😄 

The `log.*` field set in ECS is intended for details around the logging mechanism or transport itself.  For example, `log.level` can capture a syslog severity level, `log.syslog.severity` to capture the severity code, etc. Some fields being parsed in this case could be mapped to other top-level ECS field sets (e.g. mapping `username` to `user.name` field under the `user` field set). Some examples of auth log parsing from beats [here](https://github.com/elastic/beats/blob/master/filebeat/module/system/auth/ingest/pipeline.yml) show some possibilities.

Reply:
Thanks Eric, was running beats as a reference but I do not think LS has anything specific for `auth` logs.
Beats is using `system.auth` namespace (and `system.syslog`) but parses different formats - no special handling for PAM session logs. I guess this is potentially half baked on LS's end.

Anyhow the best way forward is (as suggested) to stop 'abusing' the `log.syslog.*` namespace and do `system.syslog.pam.*` here, I think I will drop the by part and just parse this into:
`Jul 14 13:36:03 precision pkexec: pam_unix(polkit-1:session): session opened for user root by (uid=1001)`
```java
"message"=>"pam_unix(polkit-1:session): session opened for user root by (uid=1001)", 
// atm this really ends up as : 
// "message"=>[
//    "Jul 14 13:36:03 precision pkexec: pam_unix(polkit-1:session): session opened for user root by (uid=1001)", 
//    "pam_unix(polkit-1:session): session opened for user root by (uid=1001)" 
// ]
// for the pattern it's best if we do match into "system.syslog.message" 
// and let the input plugin do the rename/rewrite "system.syslog.message" -> "message"
"user.name"=>"root", 
"process.program"=>"pkexec", 
"system"=>{
  "syslog"=>{
    "pam"=>{"module"=>"pam_unix", "session_state"=>"opened", "origin"=>"polkit-1:session"}
    // (previous) "pam_by" => "(uid=0)" dropped
  }
}, 
"timestamp"=>"Jul 14 13:36:03", 
"event.provider"=>"precision" // will be "host.hostname" due beats compatibility
```

Reply:
Could `process.name` (ECS) be adopted over `process.program` (non-ECS)?

Reply:
thanks, yeah sorry about that it was meant to be `process.name` - will update the PR

Reply:
Yeah I also support aligning with Beats and nesting under `system.auth.*` and such, for fields that aren't in ECS.